pennylane/qnode.py

```diff
@@ -31,6 +31,21 @@
 from pennylane.tape import JacobianTape, QubitParamShiftTape, CVParamShiftTape, ReversibleTape
 
 
+def _jax_float64_support():
+    """Checks if float64 is enabled for jax.  Returns True of float64 support enabled,
+    False if not.
+    """
+    import jax.numpy as jnp
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("error")
+        try:
+            jnp.array(0.1, dtype=jnp.float64)
+        except UserWarning:
+            return False
+    return True

```

Comment:
Nice workaround, but it feels a bit too reliant on implementation details of JAX and how it raises warnings. 

I would suggest instead simply querying the JAX configuration:

```python
>>> from jax.config import config
>>> config.x64_enabled
False
>>> config.update("jax_enable_x64", True)
>>> config.x64_enabled
True
```

Reply:
Also, I would recommend moving this to `interfaces/jax.py`, so that all JAX imports are isolated in a single place. The warning can then be raised by the tape mixin class querying `jacobian_options` dictionary